sycamore/docset.py

```diff
@@ -269,6 +270,19 @@ def extract_entity(self, entity_extractor: EntityExtractor, **kwargs) -> "DocSet
 
         entities = ExtractEntity(self.plan, entity_extractor=entity_extractor, **kwargs)
         return DocSet(self.context, entities)
+    
+    def extract_schema(self, schema_extractor: SchemaExtractor, **kwargs) -> "DocSet":
+        from sycamore.transforms import ExtractSchema
+
+        schema = ExtractSchema(self.plan, schema_extractor=schema_extractor)
+        return DocSet(self.context, schema)
+    
+    def extract_properties(self, property_extractor: PropertyExtractor, **kwargs) -> "DocSet":
```

Comment:
Is it worth having this separate from `extract_entity`? I get that they behave slightly differently, but from an external perspective they do almost the same thing. I wonder if we can unify the `PropertyExtractor` and `EntityExtractor`  classes?

Reply:
Sorry for the delay on getting back to this. My thinking here is that there a setting in which the user provides the schema themselves (so `extract_schema` is **not** needed) but still want us to extract the actual entities (so `extract_properties` is needed). The user may even want to modify the schema produced by us before using it to extract entities.